### ISSUE TITLE
`bundle package --all-platforms` causes `bundle install` to ignore `--with` and `--without`

### ISSUE THREAD

**Issue by mernen** [bundler/bundler#6074]
When the `--all-platforms` flag is set, Bundler will always try to install all packages from all groups, even if they are explicitly set to be ignored in the current machine.

Reproduction script: <https://gist.github.com/mernen/adc97b040c6310468919188791170259>

```sh
#!/bin/sh
set -ex

mkdir -p /tmp/repro-6074
cd /tmp/repro-6074

bundle env
# Bundler   1.15.4
# Rubygems  2.6.13
# Ruby      2.4.2p198 (2017-09-14 revision 59899) [x86_64-darwin16]
# Git       2.14.2
# Platform  x86_64-darwin-16
# OpenSSL   OpenSSL 1.0.2l  25 May 2017

cat > Gemfile <<EOF
source 'https://rubygems.org'
gem 'dotenv'
group :oracle do
  gem 'ruby-oci8'
end
EOF

bundle install --path vendor/bundle --without oracle

# Expected behavior: should prepare vendor/cache with ruby-oci8, but without building this gem
# Observed behavior: attempts to build ruby-oci8 (will likely fail due to missing system libraries)
bundle package --all-platforms

# Expected behavior: exact same as first invocation
# Observed behavior: attempts to build ruby-oci8 even though it's explicitly ignored
bundle install --path vendor/bundle --without oracle
```

---

In short, I don’t think `bundle package` settings should affect `bundle install` behavior.

Please note there’s nothing special about the two gems in this example; `ruby-oci8` was chosen simply because it’s very unlikely to succeed in building (as it has very specific requirements), thus highlighting why ignoring `with`/`without` can be problematic.

**Comment by colby-swandale** [comment]
Thanks! I did some digging and this seems like a symptom to a larger problem.

@indirect do you know why package is setting `options[:update] = true' in https://github.com/bundler/bundler/blob/master/lib/bundler/cli/package.rb#L32

**Comment by segiddins** [comment]
Thanks for the excellent issue! https://github.com/bundler/bundler/pull/6113 should fix it up!

**Comment by joshuaspence** [comment]
Is the fix in #6113 going to be merged anytime soon?

**Comment by segiddins** [comment]
@joshuaspence rebased it, hopefully its passing
